<launch>

  <include file="$(find igvc_description)/launch/igvc_launch.launch" />

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find igvc_control)/config/igvc_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/igvc" args="left_steering_position_controller right_steering_position_controller left_rear_velocity_controller right_rear_velocity_controller"/>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
    <remap from="/joint_states" to="/igvc/joint_states" />
  </node>

  <!-- listen to boardcomms and translate it to ros_control commands -->
  <node name="SimBoardComms" pkg="igvc_control" type="SimBoardComms.py" respawn="false" />

</launch>